src/org/labkey/test/components/react/QueryChartDialog.java

```diff
@@ -0,0 +1,330 @@
+        return new ElementCache();
+    }
+
+    @Override
+    protected ElementCache elementCache()
+    {
+        return (ElementCache) super.elementCache();
+    }
+
+    protected class ElementCache extends ModalDialog.ElementCache
+    {
+        final Input nameInput = Input(Locator.input("name"), getDriver()).findWhenNeeded(this);
+        final Checkbox sharedCheckbox = Checkbox.Checkbox(Locator.input("shared")).findWhenNeeded(this);
+
+        Locator.XPathLocator chartBuilderType = Locator.tagWithClass("div", "chart-builder-type");
+        public CHART_TYPE getSelectedChartType()
+        {
+            var selectedEl = chartBuilderType.withAttributeContaining("class", "selected")
+                    .waitForElement(this, 1500);
+            String dataName = selectedEl.getAttribute("data-name");
+            return CHART_TYPE.fromChartType(dataName);
+        }
+
+        // note: selecting a chart type will cause most select fields to go stale
+        public void selectChartType(CHART_TYPE chartType)
+        {
+            if (getSelectedChartType().equals(chartType))
+                return;
+
+            var el = chartBuilderType.withAttribute("data-name", chartType.getChartType())
+                    .waitForElement(this, 1500);
+            getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(el));
+            el.click();
+            WebDriverWrapper.waitFor(()-> getSelectedChartType().equals(chartType),
+                    "The requested chart type did not become selected", 2000);
+        }
+
+        public ReactSelect reactSelectByLabel(String label)
+        {
+            WebElement container = Locator.tag("div").withChild(Locator.tagContainingText("label", label))
+                    .waitForElement(this, 1500);
+            return ReactSelect.finder(getDriver()).find(container);
+        }
+
+        private Locator.XPathLocator previewContainerLoc = Locator.tag("div").withChild(Locator.tagWithText("label", "Preview"));
+        public WebElement previewContainer()
+        {
+            return previewContainerLoc.waitForElement(this, 1500);
+        }
+        public String grayTextPreviewInstruction()
+        {
+            return Locator.tagWithClass("div", "gray-text").waitForElement(previewContainer(), 1500).getText();
+        }
+
+        private Locator previewBodyLoc = Locator.tagWithClass("div", "chart-builder-preview-body");
+        private Locator svgLoc = Locator.tagWithClass("div", "svg-chart__chart");
+        public boolean isPreviewPresent()
+        {
+            return previewBodyLoc.existsIn(previewContainer()) && svgLoc.existsIn(previewContainer());
+        }
+
+        public WebElement svg()
+        {
+            return svgLoc.waitForElement(previewContainer(), 1500);
+        }
+
+    }
+
+    public enum CHART_TYPE{
+        Bar("bar_chart"),
+        Box("box_plot"),
+        Line("line_plot"),
+        Pie("pie_chart"),
+        Scatter("scatter_plot");
```

Comment:
(Something of a nit-pick comment)
Enum type should be upper case in Java: [Oracle Java Doc](https://docs.oracle.com/javase/tutorial/java/javaOO/enum.html)